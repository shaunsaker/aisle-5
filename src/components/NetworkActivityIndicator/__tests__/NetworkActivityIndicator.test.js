import React from 'react';
import renderer from 'react-test-renderer';

import { NetworkActivityIndicator } from '..';

describe('NetworkActivityIndicator', () => {
  const spies = [];
  const network = {
    online: {
      type: 'wifi',
    },
    offline: {
      type: 'none',
    },
  };
  const pendingTransactions = {
    empty: [],
    notEmpty: [{}],
  };

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(
        <NetworkActivityIndicator
          network={network.online}
          pendingTransactions={pendingTransactions.empty}
        />,
      );

      expect(component).toMatchSnapshot();
    });

    it('renders the offline state', () => {
      const component = renderer.create(
        <NetworkActivityIndicator
          network={network.offline}
          pendingTransactions={pendingTransactions.empty}
        />,
      );

      expect(component).toMatchSnapshot();
    });

    it('renders the loading state', () => {
      const component = renderer.create(
        <NetworkActivityIndicator
          network={network.online}
          pendingTransactions={pendingTransactions.notEmpty}
        />,
      );

      expect(component).toMatchSnapshot();
    });
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
  });
});
