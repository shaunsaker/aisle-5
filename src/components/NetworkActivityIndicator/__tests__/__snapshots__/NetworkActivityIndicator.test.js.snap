// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkActivityIndicator renders renders the loading state 1`] = `
<View
  style={Object {}}
>
  <ActivityIndicator
    animating={true}
    color="#EDE8CA"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;

exports[`NetworkActivityIndicator renders renders the offline state 1`] = `
<View
  style={Object {}}
/>
`;

exports[`NetworkActivityIndicator renders renders with minimum required props 1`] = `
<View
  style={Object {}}
/>
`;
